synapse/handlers/auth.py

```diff
@@ -504,6 +505,144 @@ def _check_local_password(self, user_id, password):
             raise LoginError(403, "", errcode=Codes.FORBIDDEN)
         defer.returnValue(user_id)
 
+    def _ldap_simple_bind(self, server, localpart, password):
+        """ Attempt a simple bind with the credentials
+            given by the user against the LDAP server.
+
+            Returns True, LDAP3Connection
+                if the bind was successful
+            Returns False, None
+                if an error occured
```

Comment:
Any reason not to just return `LDAP3Connection` or `None`? Or even throw an exception to make sure the case is handled?